src/video-player/components/runtime.tsx

```diff
@@ -0,0 +1,155 @@
+        player.src(url);
+
+        if (authoredState.captionUrl) {
+          player.addRemoteTextTrack({
+            kind: 'captions',
+            language: 'en',
+            label: 'English',
+            src: authoredState.captionUrl,
+            'default': true
+          }, false);
+        }
+      });
+
+    if (authoredState.fixedAspectRatio) {
+      player.aspectRatio(getAspectRatio(authoredState.fixedAspectRatio));
+    }
+    if (authoredState.fixedHeight) {
+      player.height(authoredState.fixedHeight);
+    }
+    player.ready(() => {
+      if (viewedTimestamp) {
+        // console.log(player.duration());
+        // console.log(viewedProgress);
+        player.currentTime(viewedTimestamp);
+      }
+    });
+
+    return () => {
+      player.dispose();
+    };
+  };
+
+  const getAspectRatio = (aspectRatio: string) => {
+    if (aspectRatio.indexOf(':') > -1) {
+      // user supplied aspect ratio as a:b, so use it
+      return aspectRatio;
+    } else {
+      // numeric - so make it into a:b format
+      const roundedAspect = Math.round(parseFloat(aspectRatio) * 100);
+      return `${roundedAspect}:100`;
+    }
+  };
+
+  const handlePlaying = (e: any) => {
+    // store current playback progress each second
+    saveStateInterval = setInterval(updateState, 1000);
+  };
+
+  const handleStop = (e: any) => {
+    clearInterval(saveStateInterval);
+    updateState();
+  };
+
+  const updateState = () => {
+    viewedTimestamp = getViewTime();
+    viewedProgress = getViewPercentage();
+    if (setInteractiveState) {
+      setInteractiveState(Object.assign({}, interactiveState, { percentageViewed: viewedProgress, lastViewedTimestamp: viewedTimestamp }));
+    }
+  };
+
+  return (
+    <div className={css.runtime}>
+      { authoredState.prompt && <div className={css.prompt}>{ authoredState.prompt }</div> }
+      <div className={`${css.videoPlayerContainer} last-viewed${viewedTimestamp}`}>
+        <div className="video-player" data-vjs-player={true}>
+          <video ref={playerRef} className="video-js vjs-big-play-centered vjs-fluid"
+            poster={authoredState.poster}
+            onPlaying={report ? undefined : handlePlaying}
+            onEnded={report ? undefined : handleStop}
+            onPause={report ? undefined : handleStop}
+          />
+        </div>
+      </div>
+      {interactiveState?.lastViewedTimestamp && <div className={css.lastViewed}>{interactiveState.lastViewedTimestamp}</div>}
+      {authoredState.credit && <div className={css.credit}>{authoredState.credit}</div>}
+      {authoredState.creditLink && <div className={css.creditLink}><a href={authoredState.creditLink} target="_blank">
+        {authoredState.creditLinkDisplayText ? authoredState.creditLinkDisplayText : authoredState.creditLink}
+      </a></div>}
+      {
+        !report &&
+        <div>
+          { submitButton }
+          { lockedInfo }
+        </div>
+      }
```

Comment:
If you refactor this interactive to use `BaseQuestionApp`, these lines won't be necessary.